Engine/TestStep.cs

```diff
@@ -859,7 +859,10 @@ public static TestStepRun RunChildStep(this ITestStep step, ITestStep childStep,
                 if(run.Verdict == Verdict.Error && throwOnBreak)
                     run.ThrowDueToBreakConditions();
             }
-
+            if (run.SuggestedNextStep == childStep.Id)
+            {
+                RunChildStep(step, childStep, throwOnBreak, currentPlanRun, currentStepRun, attachedParameters);
```

Comment:
I think we need to return the run from the recursive call here. Imagine for example a Repeat mixin which repeats a step until the verdict is `Pass`. Then we should return the TestStepRun from the attempt that succeeded.